website/docs/guides/dbt-ecosystem/databricks-guides/how-to-set-up-your-databricks-dbt-project.md

```diff
@@ -0,0 +1,145 @@
+
+1. In Databricks, click on your Databricks username in the top bar and select User Settings in the drop down.
+2. On the Access token tab, click Generate new token.
+3. Click Generate.
+4. Copy the displayed token and click Done. (don’t lose it!)
+
+
+For your development credentials/profiles.yml:
+
+1. Set your default catalog to dev.
+2. Your developer schema should be named after yourself. We recommend dbt_<first_name_initial><last_name>.
+
+By doing this, dbt will create the developer schemas if it doesn’t already exist in the **dev** catalog but can be overridden to specify sources or production targets.
+
+### Defining your dbt deployment environment
+
+Last, we need to give dbt a way to deploy code outside of development environments. To do so, we’ll use dbt [environments](https://docs.getdbt.com/docs/collaborate/environments) to define the production targets that end users will interact with.
+
+Core projects can use [targets in profiles](https://docs.getdbt.com/docs/get-started/connection-profiles#understanding-targets-in-profiles) to separate environments. [dbt Cloud environments](https://docs.getdbt.com/docs/get-started/develop-in-the-cloud#set-up-and-access-the-cloud-ide) allow you to define environments via the UI and [schedule jobs](https://docs.getdbt.com/docs/get-started/getting-started/building-your-first-project/schedule-a-job) for specific environments.
+
+Let’s set up our deployment environment:
+
+1. In Databricks, [set up your service principal’s token](https://docs.databricks.com/dev-tools/service-principals.html#use-curl-or-postman). To do this, you will need to open up your terminal.
+2. In your terminal, define 2 environment variables:
+    1. **DATABRICKS_HOST** - the url used to log into your Databricks workspace
+    2. **DATABRICKS_TOKEN** - your [personal access token](https://docs.databricks.com/dev-tools/auth.html#pat) (also used for dbt Cloud developer credentials)
+
+```bash
+export DATABRICKS_HOST="https://dbc-a1b2345c-d6e78.cloud.databricks.com"
+export DATABRICKS_TOKEN="dapi1234567890b2cd34ef5a67bc8de90fa12b"
+```
+
+3. Next, run a curl command to get the “application_id” of your service principal. You can also get this value from the Databricks workspace admin console > Service Principals > Edit
+
+```bash
+curl -X GET \
+${DATABRICKS_HOST}/api/2.0/preview/scim/v2/ServicePrincipals \
+--header "Authorization: Bearer ${DATABRICKS_TOKEN}" \
+| jq .
+```
+
+4. Then you create a JSON file with the application ID retrieved from the above command, a comment to describe what the token will be used for, and the amount of time the token will be valid (in seconds) before it expires and will need to be regenerated.
+
+```bash
+vim create-service-principal.json
+```
+
+```json
+{
+  "application_id": "<application-id>",
+  "comment": "<comment>",
+  "lifetime_seconds": 1209600
```

Comment:
This is 2 weeks - what happens when this expires? Does someone have to go and do this from scratch every 2 weeks? Or is this token going to be a one-time use as we generate the _actual_ token we need?

Reply:
If it is meant to be a long-lived token, are we allowed to set its lifetime to 999999999999999? Is there any security implication to that? We should set our default to the actual best practice

Reply:
That was the default number in Databrick's doc. I think when I tested in the UI the max was something around 3 years (might not be the same when doing it this way). 

I didn't want to change it in case Databricks selected it for security reasons. It's typically considered bad practice to have tokens like that that don't expire but realistically it would not be practical to rotate tokens for dbt Cloud service principles

Reply:
Personally, I would use the max value but I wasn't a security-minded developer :)

Reply:
>when I tested in the UI

If this is possible in the UI, why are we messing around on the command line? I thought the docs said this _had_ to be done programmatically which is the only reason I didn't complain earlier 😅

Reply:
> Personally, I would use the max value 

same

>but I wasn't a security-minded developer :)

also same

Can we update it to 1 year, and tell people to set an annual calendar reminder to renew it? This feels super janky.

Reply:
> 

You can create them in the UI for users who can log into the workspace. Service principles cannot, so it needs to be done programmatically (at least for now, I hope they add it to the admin UI).

Reply:
Removed these steps since there are too many edge cases to consider. Added a comment when directing them to the Databricks doc about the validity time